### PR TITLE
Migrate docs to mkdocs

### PR DESCRIPTION
**Pull request by arnetheduck**
`mkdocs` works with markdown similar to `mdbook` but is generally more
pleasing to the eye and has several nice UX features.

This PR does the bulk of the transition - likely, a followup would be
needed to fully make use of the extra features and navigation.

Book pages have been kept url-compatible, meaning that for the most
part, old links should continue to work!

A preview of the new book is available from https://arnetheduck.github.io/nimbus-eth2-docs/

### CODE REVIEW

File: docs/the_nimbus_book/src/quick-start.md [hunk 1]
```diff
@@ -10,13 +10,13 @@ Running a beacon node is a [worthwhile endeavor](https://vitalik.ca/general/2021
 
 The guide assumes [Ubuntu Linux](https://ubuntu.com/download/server) is being used, and therefore some familiarity with [the linux command line](https://ubuntu.com/tutorials/command-line-for-beginners) is needed.
 
-```admonish note
-To become a validator, you will first need to be running a beacon node.
-```
+!!! note
+    To become a validator, you will first need to be running a beacon node.
+
```
Comment (etan-status) [inline docs/the_nimbus_book/src/quick-start.md:15]:
```suggestion
```

File: docs/the_nimbus_book/src/quick-start.md [hunk 2]
```diff
@@ -37,9 +37,9 @@ wget https://github.com/status-im/nimbus-eth2/releases/download/v22.6.1/nimbus-e
 tar xvf nimbus-eth2_Linux_amd64_22.6.1_2444e994.tar.gz --strip-components 1 -C nimbus-eth2
 ```
 
-```admonish tip
-Advanced users looking to take advantage of hardware-specific features and optimization may wish to [build from source](./build.md) instead!
-```
+!!! tip
+    Advanced users looking to take advantage of hardware-specific features and optimization may wish to [build from source](./build.md) instead!
+
```
Comment (etan-status) [inline docs/the_nimbus_book/src/quick-start.md:41]:
```suggestion
```

File: docs/the_nimbus_book/src/voluntary-exit.md [hunk 3]
```diff
@@ -1,43 +1,38 @@
...
+    This feature is available from `v1.7.0` onwards - earlier versions relied on the now removed [JSON-RPC API](./api.md).
 
 Voluntary exits allow validators to permanently stop performing their duties, and eventually recover the deposit.
 
 Exits are subject to a wait period that depends on the length of the exit queue. While a validator is exiting, it still must perform its duties in order not to lose funds to inactivity penalities.
 
-```admonish warning
-Voluntary exits are **irreversible**. You won't be able to validate again with the same key.
+!!! warning
+  Voluntary exits are **irreversible**. You won't be able to validate again with the same key.
```
Comment (etan-status) [inline docs/the_nimbus_book/src/voluntary-exit.md]:
```suggestion
    Voluntary exits are **irreversible**. You won't be able to validate again with the same key.
```

File: docs/the_nimbus_book/src/voluntary-exit.md [hunk 4]
```diff
@@ -1,43 +1,38 @@
...
 
-You will also not be able to withdraw your funds until a future hard fork that enables withdrawals.*
-```
+  You will also not be able to withdraw your funds until a future hard fork that enables withdrawals.*
```
Comment (etan-status) [inline docs/the_nimbus_book/src/voluntary-exit.md]:
```suggestion
    You will also not be able to withdraw your funds until a future hard fork that enables withdrawals.*
```

File: docs/the_nimbus_book/src/index.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+</br>
+
+<blockquote class="twitter-tweet"><p lang="en" dir="ltr">&quot;just because it [Nimbus] is optimized to be minimally resource intensive, doesn&#39;t mean you can&#39;t run it on a server. It means that when you do run it on a server, it is consuming a lot less resources.&quot; <a href="https://t.co/F2sdZouBtD">https://t.co/F2sdZouBtD</a></p>&mdash; Nimbus (@ethnimbus) <a href="https://twitter.com/ethnimbus/status/1376836270245154817?ref_src=twsrc%5Etfw">March 30, 2021</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+</br>
+
+This book explains the ways in which you can use Nimbus to either monitor the beacon chain or become a fully-fledged validator.
+
+!!! tip
+    [The Merge](https://ethereum.org/en/upgrades/merge/) is happening soon! Bookmark our [merge readiness](./merge.md) page to stay on top of how you need to prepare.
```
Comment (etan-status) [inline docs/the_nimbus_book/src/index.md]:
```suggestion
    [The Merge 🐼](https://ethereum.org/en/upgrades/merge/) is happening soon! Bookmark our [merge readiness](./merge.md) page to stay on top of how you need to prepare.
```

**Review by etan-status — APPROVED** [review]

File: docs/Makefile [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+mkdocs/pyenv.cfg:
+	test -d mkdocs || python3 -m venv mkdocs
+	. mkdocs/bin/activate; pip install -r requirements.txt
+
+compile:
+	pip-compile requirements.in
+
+sync:
+	pip-sync requirements.txt
```
Comment (zah) [inline docs/Makefile]: Is this using the virtualenv automatically? That's not typical for other pip commands.

Comment (arnetheduck) [reply docs/Makefile]: 318ea4d8c3f88b3edef1b99f916c72f26fc7171c :+1:

**Pull request merged**